-Waddress and -Wnonnull-compare warnings when building in release mode

I am using g++ 13.3.0 on Ubuntu 24.04.3. When building in release mode (-DCMAKE_BUILD_TYPE=RelWithDebInfo), I see multiple -Waddress warnings. For example:
```
In file included from /home/pbrannan/git/glaze/include/glaze/json/json_ptr.hpp:11,
                 from /home/pbrannan/git/glaze/include/glaze/core/ptr.hpp:9,
                 from /home/pbrannan/git/glaze/include/glaze/beve/ptr.hpp:8,
                 from /home/pbrannan/git/glaze/include/glaze/beve.hpp:8,
                 from /home/pbrannan/git/glaze/include/glaze/glaze.hpp:36,
                 from /home/pbrannan/git/glaze/tests/networking_tests/jsonrpc_registry_test/jsonrpc_registry_test.cpp:4:
/home/pbrannan/git/glaze/include/glaze/json/read.hpp: In instantiation of ‘static void glz::from<10, T>::op(auto:762&&, auto:763&&, auto:764&&, auto:765) [with auto Options = glz::opts{10, true, false, true, true, false, false, false, false, 0}; auto:762 = std::__cxx11::basic_string<char> (&)(); auto:763 = glz::context&; auto:764 = const char*&; auto:765 = const char*; T = std::__cxx11::basic_string<char>()]’:
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:60:45:   required from ‘static void glz::parse<10>::op(T&&, Ctx&&, It0&&, It1) [with auto Opts = glz::opts{10, true, false, true, true, false, false, false, false, 0}; T = std::__cxx11::basic_string<char> (&)(); Ctx = glz::context&; It0 = const char*&; It1 = const char*]’
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:4607:37:   required from ‘static void glz::from<10, T>::op(auto:762&&, auto:763&&, auto:764&&, auto:765) [with auto Options = glz::opts{10, true, false, true, true, false, false, false, false, 4}; auto:762 = std::__cxx11::basic_string<char> (* const&)(); auto:763 = glz::context&; auto:764 = const char*&; auto:765 = const char*; T = std::__cxx11::basic_string<char> (*)()]’
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:302:83:   required from ‘void glz::decode_index(Value&&, auto:618&&, auto:619&&, auto:620&&, SelectedIndex&& ...) [with auto Opts = opts{10, true, false, true, true, false, false, false, false, 0}; T = static_method_api_t; long unsigned int I = 0; Value = static_method_api_t&; SelectedIndex = {}; auto:618 = context&; auto:619 = const char*&; auto:620 = const char*&]’
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:508:40:   required from ‘glz::parse_and_invoke<opts{10, true, false, true, true, false, false, false, false, 0}, static_method_api_t, static_method_api_t&, context&, const char*&, const char*&>(static_method_api_t&, context&, const char*&, const char*&)::<lambda()> [with long unsigned int I = 0]’
/home/pbrannan/git/glaze/include/glaze/util/for_each.hpp:103:45:   required from ‘constexpr void glz::visit(auto:138&&, size_t) [with long unsigned int N = 3; auto:138 = parse_and_invoke<opts{10, true, false, true, true, false, false, false, false, 0}, static_method_api_t, static_method_api_t&, context&, const char*&, const char*&>(static_method_api_t&, context&, const char*&, const char*&)::<lambda()>; size_t = long unsigned int]’
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:508:21:   [ skipping 4 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/home/pbrannan/git/glaze/include/glaze/core/read.hpp:124:24:   required from ‘glz::error_ctx glz::read(T&, auto:228&&) [with auto Opts = opts{10, true, false, true, true, false, false, false, false, 0}; T = static_method_api_t; auto:228 = std::basic_string_view<char>&]’
/home/pbrannan/git/glaze/include/glaze/rpc/jsonrpc_registry_impl.hpp:102:30:   required from ‘bool glz::jsonrpc::read_params(Value&&, auto:1346&&) [with auto Opts = glz::opts{10, true, false, true, true, false, false, false, false, 0}; Value = static_method_api_t&; auto:1346 = state&]’
/home/pbrannan/git/glaze/include/glaze/rpc/jsonrpc_registry_impl.hpp:130:47:   required from ‘static void glz::registry_impl<Opts, 30200>::register_read_write_endpoint(glz::sv, T&, RegistryType&) [with T = static_method_api_t; RegistryType = glz::registry<glz::opts{10, true, false, true, true, false, false, false, false, 0}, 30200>; auto Opts = glz::opts{10, true, false, true, true, false, false, false, false, 0}; glz::sv = std::basic_string_view<char>]’
/home/pbrannan/git/glaze/include/glaze/rpc/jsonrpc_registry_impl.hpp:152:38:   required from ‘static void glz::registry_impl<Opts, 30200>::register_endpoint(glz::sv, T&, RegistryType&) [with T = static_method_api_t; RegistryType = glz::registry<glz::opts{10, true, false, true, true, false, false, false, false, 0}, 30200>; auto Opts = glz::opts{10, true, false, true, true, false, false, false, false, 0}; glz::sv = std::basic_string_view<char>]’
/home/pbrannan/git/glaze/include/glaze/rpc/registry.hpp:249:36:   required from ‘void glz::registry<Opts, Proto>::on(T&) [with const std::string_view& root = glz::detail::empty_path; T = static_method_api_t; const std::string_view& parent = glz::detail::empty_path; auto Opts = glz::opts{10, true, false, true, true, false, false, false, false, 0}; unsigned int Proto = 30200]’
/home/pbrannan/git/glaze/tests/networking_tests/jsonrpc_registry_test/jsonrpc_registry_test.cpp:484:16:   required from here
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:4579:20: warning: the compiler can assume that the address of ‘value’ will never be NULL [-Waddress]
 4579 |                if (!value) {
      |                    ^~~~~~
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:4537:29: note: ‘value’ declared here
 4537 |       static void op(auto&& value, is_context auto&& ctx, auto&& it, auto end)
```

The same line of code also produces this warning, presumably for the same reason:
```
/home/pbrannan/git/glaze/include/glaze/json/read.hpp: In static member function ‘static void glz::from<10, T>::op(auto:762&&, auto:763&&, auto:764&&, auto:765) [with auto Options = glz::opts{10, true, false, true, true, false, false, false, false, 0}; auto:762 = std::__cxx11::basic_string<char> (&)(); auto:763 = glz::context&; auto:764 = const char*&; auto:765 = const char*; T = std::__cxx11::basic_string<char>()]’:
/home/pbrannan/git/glaze/include/glaze/json/read.hpp:4579:16: warning: ‘nonnull’ argument ‘value’ compared to NULL [-Wnonnull-compare]
 4579 |                if (!value) {
      |                ^~
```

I'm not sure the right way to fix this.  One solution is to move the `if (!value)` check inside the constexpr-ifs' then-statements, and omit the check in the else-statement.  I do not know if that has the semantics you are looking for here.

--- a/include/glaze/beve/read.hpp
+++ b/include/glaze/beve/read.hpp
@@ -1736,8 +1736,7 @@ namespace glz
       }
    };
 
-   template <nullable_t T>
-      requires(!std::is_array_v<T> && not is_expected<T>)
+   template <nullable_like T>
    struct from<BEVE, T> final
    {
       template <auto Opts>

--- a/include/glaze/beve/size.hpp
+++ b/include/glaze/beve/size.hpp
@@ -390,8 +390,7 @@ namespace glz
       }
    };
 
-   template <nullable_t T>
-      requires(!std::is_array_v<T>)
+   template <nullable_like T>
    struct calculate_size<BEVE, T>
    {
       template <auto Opts>
@@ -461,7 +460,7 @@ namespace glz
          if constexpr (std::same_as<V, hidden> || std::same_as<V, skip>) {
             return true;
          }
-         else if constexpr (is_member_function_pointer<V>) {
+         else if constexpr (is_any_function_ptr<V>) {
             return !check_write_function_pointers(Opts);
          }
          else {
@@ -528,7 +527,7 @@ namespace glz
          if constexpr (std::same_as<V, hidden> || std::same_as<V, skip>) {
             return true;
          }
-         else if constexpr (is_member_function_pointer<V>) {
+         else if constexpr (is_any_function_ptr<V>) {
             return !check_write_function_pointers(Opts);
          }
          else {

--- a/include/glaze/beve/write.hpp
+++ b/include/glaze/beve/write.hpp
@@ -1079,8 +1079,7 @@ namespace glz
       }
    };
 
-   template <nullable_t T>
-      requires(!std::is_array_v<T> && not is_expected<T>)
+   template <nullable_like T>
    struct to<BEVE, T> final
    {
       template <auto Opts, class B>

--- a/include/glaze/cbor/read.hpp
+++ b/include/glaze/cbor/read.hpp
@@ -1671,8 +1671,7 @@ namespace glz
    };
 
    // Nullable types
-   template <nullable_t T>
-      requires(!std::is_array_v<T> && not is_expected<T>)
+   template <nullable_like T>
    struct from<CBOR, T> final
    {
       template <auto Opts>

--- a/include/glaze/cbor/write.hpp
+++ b/include/glaze/cbor/write.hpp
@@ -612,7 +612,7 @@ namespace glz
          if constexpr (std::same_as<V, hidden> || std::same_as<V, skip>) {
             return true;
          }
-         else if constexpr (is_member_function_pointer<V>) {
+         else if constexpr (is_any_function_ptr<V>) {
             return !check_write_function_pointers(Opts);
          }
          else {
@@ -900,8 +900,7 @@ namespace glz
    };
 
    // Nullable types (std::optional, std::unique_ptr, std::shared_ptr)
-   template <nullable_t T>
-      requires(!std::is_array_v<T> && not is_expected<T>)
+   template <nullable_like T>
    struct to<CBOR, T> final
    {
       template <auto Opts>

--- a/include/glaze/core/common.hpp
+++ b/include/glaze/core/common.hpp
@@ -396,12 +396,6 @@ namespace glz
    template <class T>
    concept always_skipped = is_includer<T> || std::same_as<T, hidden> || std::same_as<T, skip>;
 
-   template <class T>
-   concept nullable_t = !meta_value_t<T> && !str_t<T> && requires(T t) {
-      bool(t);
-      { *t };
-   };
-
    // Detect function pointers and function references (which should not be treated as nullable)
    template <class T>
    concept is_function_ptr_or_ref =
@@ -414,7 +408,13 @@ namespace glz
    concept is_any_function_ptr = is_member_function_pointer<T> || is_function_ptr_or_ref<T>;
 
    template <class T>
-   concept nullable_like = nullable_t<T> && !is_expected<T> && !std::is_array_v<T> && !is_function_ptr_or_ref<T>;
+   concept nullable_t = !meta_value_t<T> && !str_t<T> && !is_function_ptr_or_ref<T> && requires(T t) {
+      bool(t);
+      { *t };
+   };
+
+   template <class T>
+   concept nullable_like = nullable_t<T> && !is_expected<T> && !std::is_array_v<T>;
 
    // For optional like types that cannot overload `operator bool()`
    template <class T>

--- a/include/glaze/json/read.hpp
+++ b/include/glaze/json/read.hpp
@@ -297,6 +297,14 @@ namespace glz
                skip_value<JSON>::op<Opts>(ctx, it, end);
             }
          }
+         else if constexpr (is_function_ptr_or_ref<V>) {
+            // Function pointers cannot be deserialized from JSON.
+            // When write_function_pointers is enabled the writer emits a type-name string,
+            // but there is nothing meaningful to reconstruct on read — just skip the value.
+            // When write_function_pointers is off the key is never written, so this branch
+            // is unreachable in that case.
+            skip_value<JSON>::op<Opts>(ctx, it, end);
+         }
          else {
             if constexpr (glaze_object_t<T>) {
                from<JSON, std::remove_cvref_t<V>>::template op<ws_handled<Opts>()>(
@@ -4529,8 +4537,9 @@ namespace glz
    };
 
    template <class T>
-      requires((nullable_t<T> || nullable_value_t<T>) && not is_expected<T> && not std::is_array_v<T> &&
-               not custom_read<T>)
+      requires((nullable_like<T> || nullable_value_t<T>) && not is_expected<T> && not std::is_array_v<T> &&
+               not custom_read<T>) // is_expected and is_array_v are redundant with nullable_like but needed for
+                                   // nullable_value_t
    struct from<JSON, T>
    {
       template <auto Options>

